src/de/jost_net/JVerein/io/BeitragsUtil.java

```diff
@@ -21,32 +21,109 @@
 import java.rmi.RemoteException;
 import java.util.Date;
 
+import de.jost_net.JVerein.Einstellungen;
 import de.jost_net.JVerein.keys.Beitragsmodel;
 import de.jost_net.JVerein.keys.Zahlungstermin;
 import de.jost_net.JVerein.rmi.Beitragsgruppe;
+import de.jost_net.JVerein.rmi.Mitglied;
+import de.jost_net.JVerein.util.VonBis;
+import de.willuhn.util.ApplicationException;
 
 public class BeitragsUtil
 {
   public static double getBeitrag(Beitragsmodel bm, Zahlungstermin zt, int zr,
-      Beitragsgruppe bg, Date stichtag, Date eintritt, Date austritt)
-      throws RemoteException
+      Beitragsgruppe bg, Date stichtag, Mitglied m)
+      throws RemoteException, ApplicationException
   {
     double betr = 0;
-    if (eintritt != null && eintritt.after(stichtag))
+    if (m.getEintritt() != null && m.getEintritt().after(stichtag))
     {
       return 0;
     }
-    if (austritt != null && austritt.before(stichtag))
+    if (m.getAustritt() != null && m.getAustritt().before(stichtag))
     {
       return 0;
     }
     switch (bm)
     {
       case GLEICHERTERMINFUERALLE:
-        betr = bg.getBetrag();
+        if(Einstellungen.getEinstellung().getGeburtsdatumPflicht() && bg.getHasAltersstaffel())
+        {
+          Einstellungen.getEinstellung().getAltersModel();
+          String stufen = Einstellungen.getEinstellung().getBeitragAltersstufen();
+          if(stufen != null && stufen != "")
+          {
+        	if(m.getAlter() == null)
+        		throw new ApplicationException(m.getName() + ", " + m.getVorname() + ": Geburtsdatum nicht vorhanden");
+            AltersgruppenParser ap = new AltersgruppenParser(stufen);
+            int i = -1;
+            VonBis vb = null;
+            while(ap.hasNext())
+            {
+              vb = ap.getNext();
+              if(m.getAlter() >= vb.getVon() && m.getAlter() <= vb.getBis())
+                break;
+              i++;
```

Comment:
Das i++ muss vor das if. Sonst ist der erste gefundene index -1, also um 1 zu wenig und man greift in die falsche Gruppe beim Lesen des Betrags.
Das gleiche unten bei MONATLICH.

Reply:
habe ich geändert

Reply:
So geht es auch, allerdings gibt es jetzt eine kleine Unschönheit. Wenn der AltersgruppenParser keine Gruppe liefert läufst du nicht in das while. Der Wert für i bleibt bei 0 und du greifst dann auf den Betrag für die Gruppe 0 zu. Das gibt dann wohl eine Exception. Aber in der benutzt du beim Bau der Meldung vb welches noch null ist. Damit generiert das nochmal eine Nullpointer Exception.
Da hat mir das mit dem -1 als Startwert besser gefallen weil man dann erkennt ob man überhaupt in das while läuft. Und wenn man dort vor dem if inkrementiert passt das i auch.

Reply:
Jetzt sehe ich gerade das Problem mit meinem Vorschlag. Man erkennt am i nicht ob ein Bereich gefunden wurde weil immer der letzte übrig bleibt und dann der Betrag davon genommen wird.
Da ist dann wohl dein Weg besser.
Ob das while durchlaufen wurde könnte man über vb = null prüfen.

Reply:
Jetzt habe ich wohl Unsinn geredet. Du prüft ja vorher ob stufen vorhanden sind. Also durchläuft man das while immer.

Reply:
Vielleicht noch das, wenn nach der while Schleife das i gleich der Anzahl von ap Einträgen ist wurde kein Bereich für das Alter gefunden. Da passt dann die Fehlermeldung im catch nicht ganz.

Reply:
Ich glaube es macht Sinn diesen Fall abzufangen und eine entsprechende Meldung auszugeben. Es gibt einen Fall, dass es nicht zur nachfolgenden Exception kommt.
Das Szenario wäre so. Man erzeugt z.B. 5 Zeitbereiche, erzeugt eine Beitragsgruppe und konfiguriert die Beiträge. Später reduziert man die Bereiche auf 4. Wie ich verstanden habe wird da der fünfte Eintrag nicht gelöscht sondern nur nicht mehr angezeigt und verwendet.
Kommt es jetzt aber hier zu dem Fall, dass für das Alter kein Bereich gefunden wird, steht i beim nächsten Eintrag der in diesem Fall existiert obwohl er ungültig ist. Der wird dann als Beitrag genommen. Es sollte aber eine Fehlermeldung geben.

Reply:
ich habe jetzt einen neuen int nummer eingeführt wie ich es vorher schon mal hatte. so sollten jetzt alle fälle abgedeckt sein